test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -213,13 +214,16 @@ private Node newNode() {
         if (addMockHttpTransport()) {
             plugins.add(MockHttpTransport.TestPlugin.class);
         }
-        Node build = new MockNode(settings, plugins, forbidPrivateIndexSettings());
-        try {
-            build.start();
-        } catch (NodeValidationException e) {
-            throw new RuntimeException(e);
-        }
-        return build;
+        Node node = new MockNode(settings, plugins, forbidPrivateIndexSettings());
+        bootstrapNodes(true,
+            () -> {
+                try {
+                    node.start();
+                } catch (NodeValidationException e) {
+                    throw new RuntimeException(e);
```

Comment:
I know this was here before, but why `RuntimeException` and not `AssertionError` here?

Reply:
We use the same pattern in InternalTestCluster, and I think the better change would be to make `NodeValidationException` a non-checked exception (so that this wrapping is not needed).